core/src/test/scala/integration/kafka/api/SaslPlainSslEndToEndAuthorizationTest.scala

```diff
@@ -16,12 +16,24 @@
   */
 package kafka.api
 
+import java.io.File
 import org.apache.kafka.common.protocol.SecurityProtocol
+import org.apache.kafka.common.config.SaslConfigs
+import kafka.utils.JaasTestUtils
 
 class SaslPlainSslEndToEndAuthorizationTest extends EndToEndAuthorizationTest {
   override protected def securityProtocol = SecurityProtocol.SASL_SSL
   override protected def kafkaClientSaslMechanism = "PLAIN"
   override protected def kafkaServerSaslMechanisms = List("PLAIN")
   override val clientPrincipal = "testuser"
   override val kafkaPrincipal = "admin"
+
+  // Use JAAS configuration properties for clients so that dynamic JAAS configuration is also tested by this set of tests
+  override protected def setJaasConfiguration(mode: SaslSetupMode, serverMechanisms: List[String], clientMechanisms: List[String],
+      serverKeytabFile: Option[File] = None, clientKeytabFile: Option[File] = None) {
+    super.setJaasConfiguration(mode, kafkaServerSaslMechanisms, List()) // create static config without client login contexts
```

Comment:
I wonder if we should call `setConfiguration` with a `Configuration` instance that throws an exception when any of its implemented methods is invoked. I think it would make it easier to diagnose if, for whatever reason, we ended up relying on the singleton `Configuration`by mistake. What do you think?

If we do that here, there's at least one other place where it would make sense to use it too (there may be more as I'm still going through the code).